### ISSUE TITLE
about:blank pages seem to be added to history if we intent out to YouTube

### ISSUE THREAD

**Issue by dan-alcantara-neeva** [neevaco/neeva-android#807]
Seems like a visit to about:blank can be committed if we fire an Intent out to another app.  Should probably stop that.

Version 0.74.1

![Screenshot_20220718-091137](https://user-images.githubusercontent.com/96092902/179555264-09670984-6666-4162-b5e2-fae8024e1c19.png)

**Comment by Kokobe** [comment]
Seems that this isn't an issue for WebLayer Shell and `weblayer` is creating an `about:blank` tab when launching an external intent. 

This only happens **sometimes**:
* @dan-alcantara-neeva has observed it happening with opening up Youtube
* I've observed it when we open the "MENTIONED BY" reddit links on neeva search

**Comment by Kokobe** [comment]
Chased it down to `RenderFrameImpl::OpenUrl()` in [content/renderer/render_frame_impl.cc](https://source.chromium.org/chromium/chromium/src/+/main:content/renderer/render_frame_impl.cc;l=5444;drc=8f243f4c58be1a3ffdfd378978d8f26f5d7ca997)

The new tab behavior is decided by the `disposition` in the params, see:
```
params->disposition = NavigationPolicyToDisposition(policy);
```

Now looking for what logic decides the navigation policy...

**Comment by Kokobe** [comment]
Followed `BeginNavigation()` to `blink` inside [frame_loader.cc](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/loader/frame_loader.cc;l=854;drc=3ffc2f307ac29a2a3467237538e785af59bf1b12). Looked at `request.GetNavigationPolicy()` and that is set to `kNavigationPolicyCurrentTab` by default (see [here](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/loader/frame_load_request.h;drc=3ffc2f307ac29a2a3467237538e785af59bf1b12;l=204)) 

The parameter of interest in `content/renderer/render_frame_impl.cc` is `request.GetNavigationPolicy()` which is translated to a `disposition` in the call stack by [content/renderer/render_frame_impl.cc](https://source.chromium.org/chromium/chromium/src/+/main:content/renderer/render_frame_impl.cc;l=5396;drc=3ffc2f307ac29a2a3467237538e785af59bf1b12)

**Comment by Kokobe** [comment]
Found [SetNavigationPolicy()](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/loader/frame_load_request.h;drc=3ffc2f307ac29a2a3467237538e785af59bf1b12;bpv=1;bpt=1;l=87) which is being used by `HandleClick()` in [third_party/blink/renderer/core/html/html_anchor_element.cc](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/html_anchor_element.cc;bpv=1;bpt=1;l=393)

The policy is determined by the event and `NavigationPolicyFromEvent` in [third_party/blink/renderer/core/loader/navigation_policy.cc](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/loader/navigation_policy.cc;l=133;drc=3ffc2f307ac29a2a3467237538e785af59bf1b12) has that logic

**Comment by Kokobe** [comment]
So it turns out that both WebLayerShell and our debug build have the same `NavigationPolicy` from `Blink`: `kNavigationPolicyCurrentTab` when clicking on the MENTIONED BY links on neeva.com...

This is weird because when we do it, we get a callback for `onNewTab()` with the about:blank page

**Comment by Kokobe** [comment]
Agh my comments here were actually unrelated to fixing this issue... because the issue was already fixed by #1050
